client/src/Views/Home/Prospects/Prospects.js

```diff
@@ -1,16 +1,176 @@
-import React from "react";
-import { Grid, Typography, withStyles } from "@material-ui/core";
+import React, { Component } from "react";
+import { Grid, withStyles, Typography } from "@material-ui/core";
+import { Route } from "react-router-dom";
+import ProspectSidebar from "./ProspectSidebar";
+import ProspectDashboardHeader from "./ProspectDashboardHeader";
+import Box from '@material-ui/core/Box';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import { getProspectData } from "Utils/api";
+import AuthUserContext from "Components/Session/AuthUserContext";
+import styles from "Components/Table/ProspectTableStyles";
+import CloudIcon from '@material-ui/icons/Cloud';
+import ProspectTableCheckbox from 'Components/Checkbox/ProspectTableCheckbox';
 
-const styles = () => ({
-  root: {
-    flexGrow: 1
+
+class Prospects extends Component {
+  
+  constructor(props) {
+    super(props);
+    this.state = {
+      prospects : [],
+      filteredProspects: [],
+      user : {},
+      checked : false,
+      errors: {},
+      loggedIn: localStorage.getItem("token") ? true : false
+    }
+    this.handleFieldChange = this.handleFieldChange.bind(this);
+  }
+  
+
+  handleClick = (event, id) => {
+    console.log(event);
+  }
+  
+
+  componentDidMount = async () => {
+    try {
+      this.state.user = this.context.user;
+      const res = await getProspectData(this.state.user.id);
+      this.setState({
+        prospects: res.data,
+        filteredProspects : res.data
+      });
+    } catch(error) {
+      this.setState({
+        errors: { error }
+      });
+    }
+  }
```

Comment:
Did you use a class here instead of a functional component because you wanted componentDidMount()? I faced the same issue last week, but found you can achieve the same in a functional component using useEffect() and passing it an empty array. You can't make useEffect async though so you'd have to put an async function inside useEffect() and call it after declaring. Like:

```
useEffect( () => {
  const asyncFunction = async () => {
    //do async stuff
  }
asyncFunction()
}, [])
```